File: JobRecruitment/src/main/webapp/src/app/model/Filter.ts

export class Filter{
  title: string;
  jobType : string;
  remote: boolean;
  minExperience: string;
  minCompensation: number;
  devType: string;
  techs: string;
  availablePos: number;


  constructor(title: string, jobType: string, remote: boolean, minExperience: string, minCompensation: number, devType: string, techs: string, availablePos: number) {
    this.title = title;
    this.jobType = jobType;
    this.remote = remote;
    this.minExperience = minExperience;
    this.minCompensation = minCompensation;
    this.devType = devType;
    this.techs = techs;
    this.availablePos = availablePos;
  }
}

File: JobRecruitment/src/main/java/bachelor/thesis/job_recruitment/core/model/Filter.java
package bachelor.thesis.job_recruitment.core.model;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
public class Filter {
    private String jobType;
    private Boolean remote;
    private String minExperience;
    private Integer minCompensation;
    private String devType;
    private String techs;
    private Integer availablePos;
}
